### PR TITLE
Prepare symex for lazy loading

### PR DESCRIPTION
**Pull request by smowton**
This is a subset of https://github.com/diffblue/cbmc/pull/1759, see that PR for all discussion.

### CODE REVIEW

**Comment by smowton** [comment]
Also see #1759 for how this will all be tested -- when the driver program modifications are in, all of the cbmc-java tests will be altered to exercise symex-driven lazy loading as well as conventional program loading.

File: src/goto-symex/symex_main.cpp
Comment (peterschrammel) [inline src/goto-symex/symex_main.cpp]: analyze

File: src/goto-symex/symex_main.cpp
Comment (peterschrammel) [inline src/goto-symex/symex_main.cpp]: This should become an invariant with https://github.com/diffblue/cbmc/pull/1843

**Comment by smowton** [comment]
@tautschnig here's the symex-modifications-only part of the lazy loading PR

File: src/goto-symex/symex_main.cpp
Comment (smowton) [reply src/goto-symex/symex_main.cpp]: Just making a test-gen bump for that PR, since it's a bit dangerous that it might move block goals around and mess up all the tests. If it doesn't suggest we merge that then this.

**Comment by tautschnig** [comment]
Thanks a lot for factoring out. I'd strongly support merging #1843 first and have this adjusted properly.

**Comment by smowton** [comment]
@peterschrammel @tautschnig any objections to merging this once Travis passes and then submitting a cleanup after #1843 lands? This way we can avoid some CI round-trips. I ask since #1843 might have to wait for #1835, and there are comments to address there from you and @thk123

**Comment by tautschnig** [comment]
I'd prefer if #1843 landed first, but I don't think my opinion should count here: To me, this really depends on how quickly #1843 can be merged, which in turn depends on whether agreement on #1843 vs. #1835 has been reached, and that's not something I expect to have insight into.

File: src/goto-symex/symex_main.cpp
Comment (tautschnig) [inline src/goto-symex/symex_main.cpp:240]: I have filed #1847 to get this cleaned up at a later stage.

**Review by tautschnig — APPROVED** [review]
I'm assuming that either this PR will land after #1843 and will be cleaned up beforehand, or else #1843 will take care of cleaning up.

**Review by martin-cs — APPROVED** [review]
I'm not opposed to the changes in analyses/ but am not an owner for the rest.

**Comment by smowton** [comment]
Rebased atop 1843.

**Review by peterschrammel — APPROVED** [review]
LGTM, modulo CI passing

**Comment by smowton** [comment]
TG bump: https://github.com/diffblue/test-gen/pull/1505

**Comment by tautschnig** [comment]
FWIW: I've restarted the Travis job, not sure why Travis is starting to act up again.

**Comment by smowton** [comment]
@tautschnig per https://www.traviscistatus.com/ there is some ongoing incident.

**Pull request merged**
